app/grandchallenge/core/forms.py

```diff
@@ -51,3 +52,45 @@ def __init__(self, *args, **kwargs):
 
     class Meta:
         fields = ("status", "rejection_text")
+
+
+class CreateUniqueTitleFormMixin:
+    """
+    Form mixing creating an item with a unique title.
+
+    Base class should have the `Meta.model` and `instance` attributes
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.fields = {
+            "title": CharField(
+                required=False,
+                initial=self.instance and self.instance.title or "",
+                max_length=self.Meta.model._meta.get_field("title").max_length,
+            ),
+            **self.fields,
```

Comment:
Why are you resetting all fields here, not just the title?

Reply:
The order of the form is dependent on the order of the keys in the dict. I've made it more explicit by calling `Form.order_fields`, but in essence it does the same.